### PR TITLE
Add support for 'on_hold' job status

### PR DESCRIPTION
**Pull request by piotrmarczydlo**
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [n/a] Examples have been added for any significant new features
- [n/a] README has been updated, if necessary

### Motivation, issues

The workflow can use the 'Manual approval' feature which returns 'status: "on_hold"', currently the Orb is not supporting it and causes infinite loop (screenshot below).
Zrzut ekranu 2020-01-30 o 20 57 47

Related also to issue: https://github.com/CircleCI-Public/sumologic-orb/issues/17

### Description

Added 'if' statement in the status checking

### CODE REVIEW

**Review by michael-nauto — APPROVED** [review]

**Comment by KyleTryon** [comment]
Hello @piotrmarczydlo,

My concern for this change is we should not wait for on-hold jobs for two main reasons:

1.  Credits are consumed while the workflow collector job is running, which means that you would be spending money for each minute the job is on-hold.

2. After 5 hours the job will be canceled. If the on-hold job is not responded to within 5 hours this will result in a failed pipeline, of which no data could be recorded.

The options we see to get around this issue until a better system exists are:

1.  End the data stream at the manual approval step, count it as if the workflow had finished at this point. This would of course mean we lose the data of any subsequent job of the manual approval.

2. End the stream at the on-hold job and _do not_ send the data. This ensures you wont receive any partial data but also means that on-hold jobs would be incompatible with this orb.

3. Allow the job to use credits while waiting for an approval, but include a timeout parameter. This means that we could instruct the orb to shut down after X minutes/hours gracefully so that we don't get caught up in the 5 hour time limit and can instruct the job to end at this point either sending or not sending the data. This would mean you won't run into failure conditions for not approving a job, but it also means you are going to be spending credits equal to the amount of time in the timeout parameter. This option is not recommended.

It is my opinion that option two makes the most sense given the technical limitations at the current time. In the future, we will have an alternative method of sending this data that will solve for all situations.

**Comment by piotrmarczydlo** [comment]
Thank you for the response. 
So if you look in PR it's an exact solution no. 2 which you have proposed.
When Orb finds 'on_hold' job it skips it.

**Comment by KyleTryon** [comment]
Hello @piotrmarczydlo,

Has this been tested?

Here is some sample output from a workflow with an on-hold job:

```
{
"next_page_token": null,
"items": [
{
"dependencies": [],
"job_number": 302,
"id": "443c8c45-e52d-416f-bec3-be0566d8d8bd",
"started_at": "2020-02-06T21:49:25Z",
"name": "build",
"project_slug": "gh/KyleTryon/CircleCI-Metadata-Build",
"status": "success",
"type": "build",
"stopped_at": "2020-02-06T21:50:08Z"
},
{
"started_at": null,
"name": "test",
"project_slug": "gh/KyleTryon/CircleCI-Metadata-Build",
"type": "approval",
"status": "on_hold",
"id": "1b2dd116-7c5d-474d-879b-78b6c5ee8f7b",
"dependencies": [
"443c8c45-e52d-416f-bec3-be0566d8d8bd"
]
},
{
"started_at": null,
"name": "deploy",
"project_slug": "gh/KyleTryon/CircleCI-Metadata-Build",
"type": "build",
"status": "blocked",
"id": "69a76028-75bf-49ea-b7d4-4940f5b6ce7a",
"dependencies": [
"1b2dd116-7c5d-474d-879b-78b6c5ee8f7b"
]
}
]
}
```

In this example, the `deploy` job which comes after the manual approval has the status of `blocked` until the on-hold job is complete.

The `blocked` status would need to also be accounted for in this case. Perhaps we could add add this to the existing check you have here and skip it as well. I believe in its current state the blocked job would keep our loop running.

There are other reasons a job could be blocked as well, such as not having enough credits in the account. In that case, I think it would also be appropriate to respond in a similar way so we should be safe.

**Comment by piotrmarczydlo** [comment]
updated as suggested

**Review by KyleTryon — APPROVED** [review]
Great 👍 thank you for this change. We will push this up immediately. You may need to update your orb version once this is merged. We will respond with the current version momentarily.

**Comment by KyleTryon** [comment]
circleci/sumologic@1.0.5 published

**Pull request merged**
